platforms/allwinner-d1/core/src/clint.rs

```diff
@@ -0,0 +1,54 @@
+use d1_pac::CLINT;
+
+/// Core Local Interruptor (ClINT) interface
+pub struct Clint {
+    clint: CLINT,
+}
+
+impl Clint {
+    /// Create a new `Clint` from the [`CLINT`](d1_pac::CLINT) peripheral
+    pub fn new(clint: CLINT) -> Self {
+        Self { clint }
+    }
+
+    /// Release the underlying [`CLINT`](d1_pac::CLINT) peripheral
+    pub fn release(self) -> CLINT {
+        self.clint
+    }
+
+    pub unsafe fn summon() -> Self {
+        Self {
+            clint: d1_pac::Peripherals::steal().CLINT,
+        }
+    }
+
+    /// Get the (machine) time value.
+    ///
+    /// Note that the CLINT of the C906 core does not implement
+    /// the `mtime` register and we need to get the time value
+    /// with a CSR, which the `riscv` crate implements for us.
+    pub fn get_mtime(&self) -> usize {
+        riscv::register::time::read()
```

Comment:
nit, take it or leave it: this comment feels like it's describing the implementation of this method more than its behavior, so i might move it into the function body:
```suggestion
    pub fn get_mtime(&self) -> usize {
        // Note that the CLINT of the C906 core does not implement
        // the `mtime` register and we need to get the time value
        // with a CSR, which the `riscv` crate implements for us.
        riscv::register::time::read()
```